launcher.py

```diff
@@ -0,0 +1,358 @@
+import argparse
+import atexit
+import json
+import os
+import platform
+import subprocess
+import sys
+import time
+import webbrowser
+from pathlib import Path
+
+CONFIG_FILE = os.path.join("config.json")
+venv_name = "virtual_env"  # For downloading python packages
+venv_dir = Path.cwd() / venv_name  # Create the path for the virtual environment
+
+
+def execute_command(command, cwd=None):
+    """
+    Executes a shell command.
+
+    Args:
+        commands (str): The command to execute.
+
+    Raises:
+        subprocess.CalledProcessError: If any command fails.
+    """
+    subprocess.check_call(command.split(), cwd=cwd, stdout=subprocess.DEVNULL, stderr=subprocess.DEVNULL)
+
+
+class Dependency:
```

Comment:
Could be more descriptive here, e.g DependencyManager as it seems to mainly manage deps?